src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -1090,7 +1092,10 @@ private void createDependencyReducedPom( Set<String> artifactsToRemove )
             origDeps = transitiveDeps;
         }
 
-        Model model = project.getOriginalModel();
+        final Model model = project.getFile() == null
+                            ? project.getOriginalModel().clone()
+                            : new MavenXpp3Reader().read( new FileInputStream( project.getFile() ) );
```

Comment:
Possible resource leak?

Reply:
@jglick how to leak the resource?

Reply:
the inputstream will not be closed but the correct way to read a pom is org.apache.maven.model.io.ModelReader anyway

Reply:
thx, let me fix it

Reply:
I think we can blame "history" there, xpp3 is kind of old these days ;).

Reply:
ok, now we use ModelReader to parse the pom